docs/reference/method/MongoDBGridFSBucket-rename.txt

```diff
@@ -31,4 +31,51 @@ Errors/Exceptions
 .. include:: /includes/extracts/error-gridfs-filenotfoundexception.rst
 .. include:: /includes/extracts/error-driver-runtimeexception.rst
 
-.. todo: add examples
+Examples
+--------
+
+.. code-block:: php
+
+   <?php
+
+   $bucket = (new MongoDB\Client)->test->selectGridFSBucket();
+
+   $stream = fopen('php://temp', 'w+b');
+   fwrite($stream, "foobar");
+   rewind($stream);
+
+   $id = $bucket->uploadFromStream('a', $stream);
+
+   $fileDocument = $bucket->getFilesCollection()->findOne(
+       ['_id' => $id],
+       ['projection' => ['filename' => 1, '_id' => 0]]
+   );
+
+   var_dump($fileDocument);
+
+   $bucket->rename($id, 'b');
+
+   $fileDocument = $bucket->getFilesCollection()->findOne(
```

Comment:
Borrowing from https://github.com/mongodb/mongo-php-library/pull/527/files#diff-62aa246a33b6cdffb4db5f82c2145063, I think a more helpful example might be:

```
var_dump(stream_get_contents($bucket->openDownloadStreamByName('b')));
```

Then you can show that the expected output is `string(6) "foobar"`.

With that change, you can probably omit dumping the original file document as well. It should be sufficient to show that we uploaded the file as "a", renamed, and downloaded it as "b".